x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/openshiftai/rerank/OpenShiftAiRerankModelTests.java

```diff
@@ -35,4 +42,66 @@ public static OpenShiftAiRerankModel createModel(
             new DefaultSecretSettings(new SecureString(apiKey.toCharArray()))
         );
     }
+
+    public void testOverrideWith_SameParams_KeepsSameModel() {
+        testOverrideWith_KeepsSameModel(buildTaskSettingsMap(2, true));
+    }
+
+    public void testOverrideWith_EmptyParams_KeepsSameModel() {
+        testOverrideWith_KeepsSameModel(buildTaskSettingsMap(null, null));
+    }
+
+    private static void testOverrideWith_KeepsSameModel(Map<String, Object> taskSettings) {
+        var model = createModel("url", "api_key", "model_name", 2, true);
+        var overriddenModel = OpenShiftAiRerankModel.of(model, taskSettings);
+
+        assertThat(overriddenModel.getTaskSettings().getTopN(), is(2));
+        assertThat(overriddenModel.getTaskSettings().getReturnDocuments(), is(true));
```

Comment:
These assertions can be replaced with an assertion that the model returned is the same instance as the original model:
```
assertThat(overriddenModel, is(sameInstance(model)));
```